lib/authorizer/jwt.js

```diff
@@ -0,0 +1,221 @@
+            {
+                property: 'Authorization',
+                type: 'header'
+            },
+            {
+                property: '*',
+                type: 'url.param'
+            }
+        ]
+    },
+
+    /**
+     * Initializes an item (extracts parameters from intermediate requests if any, etc)
+     * before the actual authorization step
+     *
+     * @param {AuthInterface} auth -
+     * @param {Response} response -
+     * @param {AuthHandlerInterface~authInitHookCallback} done -
+     */
+    init: function (auth, response, done) {
+        done();
+    },
+
+    /**
+     * Checks the item, and fetches any parameters that are not already provided.
+     *
+     * @param {AuthInterface} auth -
+     * @param {AuthHandlerInterface~authPreHookCallback} done -
+     */
+    pre: function (auth, done) {
+        return done(null, true);
+    },
+
+    /**
+     * Verifies whether the auth succeeded
+     *
+     * @param {AuthInterface} auth -
+     * @param {Response} response -
+     * @param {AuthHandlerInterface~authPostHookCallback} done -
+     */
+    post: function (auth, response, done) {
+        done(null, true);
+    },
+
+    /**
+     * Signs the request
+     *
+     * @param {AuthInterface} auth -
+     * @param {Request} request -
+     * @param {AuthHandlerInterface~authSignHookCallback} done -
+     */
+    sign: function (auth, request, done) {
+        // bail out - invalid algorithm
+        if (!ALGORITHMS_SUPPORTED[auth.get(ALGORITHM)]) {
+            return done();
+        }
+
+        try {
+            let signKey = null;
+
+            const rawHeader = auth.get(HEADER),
+                header = _.isString(rawHeader) ? JSON.parse(rawHeader) : rawHeader,
+                // must stringify payload - else by default 'iat claim' will be auto added in payload
+                rawPayload = auth.get(PAYLOAD),
+                payload = _.isString(rawPayload) ? rawPayload : JSON.stringify(rawPayload),
+                options = {
+                    algorithm: auth.get(ALGORITHM),
+                    // treat root level alg as source of truth & remove alg from header to avoid conflict
+                    header: header && _.omit(header, [HEADER_ALGORITHM])
+                };
+
+            // HS Algorithms use secret for token generation
+            if (HS_ALGORITHMS[options.algorithm]) {
+                const secret = auth.get(SECRET),
+                    isBase64 = auth.get(IS_SECRET_BASE_64_ENCODED);
+
+                signKey = isBase64 ? Buffer.from(secret, BASE64).toString(ASCII) : secret;
+            }
+
+            // RS,PS,ES Algorithms use private key for token generation
+            else {
+                const privateKey = auth.get(PRIVATE_KEY),
+                    passphrase = auth.get(PASSPHRASE);
+
+                signKey = passphrase ? {
+                    key: privateKey,
+                    passphrase: passphrase
+                } : privateKey;
```

Comment:
What happens if `privateKey` is not set? What error message (if any) would be sent?